hw_pandas/Hw_pandas_pain.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import pandas as pd
+import re
+import seaborn as sns
+import matplotlib.pyplot as plt
+from matplotlib.font_manager import FontProperties
+import matplotlib.ticker as tck
+from matplotlib import rc,rcParams
+
+
+# Задание 1
+
+def read_gff(file):
+    gff_table = pd.read_csv(file, sep='\t', header = 0, names= ['chromosome', 
+                                                           'source', 'type',
+                                                           'start', 'end',
+                                                           'score', 'strand',
+                                                           'phase', 'RNA type'])
+    return gff_table
+
+def read_bed6(file):
+    bed_table = pd.read_csv(file, sep='\t', names = ['chromosome','start',
+                                                     'end','name','score',
+                                                     'strand'])
+    return bed_table
+
+read_bed6('alignment.bed')
+
+read_gff('rrna_annotation.gff')
+
+
+gff_table = read_gff('rrna_annotation.gff')
+gff_table['RNA type'] = gff_table['RNA type'].str.extract(r'([0-9]+S )')
+gff_table
+
+
+# Сделайте таблицу, где для каждой хромосомы (на самом деле это не хромосомы, а 
+# референсные геномы) показано количество рРНК каждого типа
+
+gff_table = read_gff('rrna_annotation.gff')
+counted_RNA = gff_table.groupby(['chromosome','RNA type']).agg({'RNA type': 'count'})
+counted_RNA
+
+
+# Постройте barplot, отображающий эти данные (картинка rRNA_barplot)
+
+counted_RNAs = gff_table.groupby(['RNA type','chromosome']).agg({'chromosome': 'count'})
+
+counted_RNAs.unstack(0).plot.bar(figsize=(20,10))
+
+# Выведите таблицу, содержащую исходные записи об рРНК полностью вошедших в сборку 
+
+gff_table = read_gff('rrna_annotation.gff')
+bed_table = read_bed6('alignment.bed')
+
+merged = pd.merge(gff_table, bed_table , how='outer', on=['chromosome'])
+
+merged_new = merged[(merged['end_x'] <= merged['end_y']) & (merged['start_x'] >= merged['start_y'])]
```

Comment:
Рабочая схема. Мне правда больше нравится через `query`, выглядит чуть лаконичнее
```suggestion
merged_new = merged.query("start_x >= start_y and end_x <= end_y")
```